iahx/views/citing-document.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+use Symfony\Component\HttpFoundation\Request;
+
+$app->get('citing-documents/{id}/', function (Request $request, $id) use ($app, $DEFAULT_PARAMS, $config) {
+
+    $lang = $request->get("lang");
+    if (! isset($lang) or ! in_array($lang, array('pt', 'es', 'en'))) {
+        $lang = $DEFAULT_PARAMS['lang'];
+    }
+
+    global $texts;
+    $texts = parse_ini_file(TRANSLATE_PATH . $lang . "/texts.ini", true);
+
+    $view = $request->get("view");
+    $site = $DEFAULT_PARAMS['defaultSite'];
+    $col = $DEFAULT_PARAMS['defaultCollection'];
+    $count = 1;
+    $output = "site";
+
+    $q = $id;
+
+    $dia = new Dia($site, $col, $count, $output, $lang);
+    $dia_response = $dia->search($q);
+
+    $result = json_decode($dia_response, true);
+
+    $citation = $result['diaServerResponse'][0]['response']['docs'][0];
```

Comment:
@rafaelpezzuto, o que acontece se `$q` for um identificador desconhecido? Haverá o índice `['docs'][0]`?

Reply:
Fazendo algumas verificações constatei que `$citation` fica como `NULL`. E na tela nada é apresentado.

Para garantir uma array limpa de sujeiras, adicionei uma verificação de `$citation`... apenas nos casos em que não for `NULL` é que a variável `output_array` é preenchida.